### ISSUE TITLE
GCC 8 Regression: Descriptor adjustment breaks get-put-allocatable-comp test

### ISSUE THREAD

**Issue by rouson** [sourceryinstitute/OpenCoarrays#518]
[issue stat img]: https://img.shields.io/issuestats/i/github/sourceryinstitute/OpenCoarrays.svg?style=flat-square

| Avg response time              |
|:------------------------------:|
| ![Issue Stats][issue stat img] |

# Defect/Bug Report #

Changes to the descriptors in gfortran [svn commit r257065]  caused several test failures in OpenCoarrays, all but one of which have been fixed.  This issue documents that the current failure of the [get-put-allocatable-comp] test stems from the aforementioned gfortran commit.  

- OpenCoarrays Version: `2.0.0-rc1-16-g07fcba9`
- Fortran Compiler: gfortran 8.0.1 20180314 (trunk)
- C compiler used for building lib: gfortran 8.0.1 20180314 (trunk)
- Installation method: `./develeloper-scripts/gcc-trunk-install.sh`
- Output of `uname -a`: `Linux sourcery-VirtualBox 4.13.0-16-generic #19-Ubuntu SMP Wed Oct 11 18:35:14 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`
- MPI library being used: MPICH 3.2
- Machine architecture and number of physical cores: Sourcery Institute virtual machine on 2.7 GHz Intel Core i7 with 4 virtual cores
- Version of CMake: `3.10.0`

## Observed Behavior ##
Test failure.

## Expected Behavior ##
Test passing.

## Steps to Reproduce ##
```bash
git clone https://github.com/sourceryinstitute/opencoarrays
cd opencoarrays
./developer-scripts/gcc-trunk-install.sh -r <packages_root>
source <packages-root>/opencoarrays/2.0.0/setup.sh
cd prerequisites/builds/opencoarrays/2.0.0
ctest --timeout 5 --output-on-failure
```

[links]:#
[Google]: https://www.google.com
[Stack Overflow]: https://stackoverflow.com
[fenced codeblocks]: https://help.github.com/articles/creating-and-highlighting-code-blocks/
[GFortran bug tracker]: https://gcc.gnu.org/bugzilla/
[Intel MPI]: https://software.intel.com/en-us/intel-mpi-library
[MPT]: http://www.sgi.com/products/software/sps.html
[MVAPICH]: http://mvapich.cse.ohio-state.edu
[MPICH]: https://www.mpich.org
[Open-MPI]: https://www.open-mpi.org
[CONTRIBUTING.md]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/CONTRIBUTING.md
[Helping Out]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/CONTRIBUTING.md#helping-out
[official release]: https://github.com/sourceryinstitute/OpenCoarrays/releases
[CMake install]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/INSTALL.md#cmake-scripts
[homebrew]: https://brew.sh
[issues page]: https://github.com/sourceryinstitute/OpenCoarrays/issues
[mailing list]: https://groups.google.com/forum/#!forum/opencoarrays
[install.sh]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/install.sh
[svn commit r257065]: https://gcc.gnu.org/viewcvs/gcc?view=revision&revision=257065
[get-put-allocatable-comp]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/src/tests/regression/reported/issue-422-send-get.F90

**Comment by zbeekman** [comment]
I have confirmed this test failure. Using MPICH master and GCC trunk as of today 2018-03-23 I see the following test failures:

```
98% tests passed, 2 tests failed out of 83

Total Test time (real) =  14.54 sec

The following tests FAILED:
	 17 - alloc_comp_multidim_shape (Failed)
	 56 - get-put-allocatable-comp (Timeout)
Errors while running CTest
```

**Comment by rouson** [comment]
@scrasmussen A GCC 8 release candidate has been [posted]  in preparation for the actual release as soon as next week.  When we have the OpenCoarrays teleconference this Friday, let's make your work on this issue the first item on the agenda.  Please plan to share screen and walk us through what you're seeing so far.  An old saying in the open-source world is that "any bug is shallow with a sufficient number of eyeballs on it."  Toward that end, I'll also ask if @sfilippone can join the call this Friday.  It would be great to have this issue resolved when GCC 8 releases.  


 [posted]: https://gcc.gnu.org/pub/gcc/snapshots/8.0.1-RC-20180425/

**Comment by vehre** [comment]
Fixed by pull request #528 .

**Comment by rouson** [comment]
I just merged PR #528, which fixes this issue as @vehre notes above so I'm closing this issue.
